pymodbus/client/base.py

```diff
@@ -136,7 +135,7 @@ def idle_time(self) -> float:
             return 0
         return self.last_frame_end + self.silent_interval
 
-    def execute(self, request: ModbusRequest | None = None) -> ModbusResponse:
+    def execute(self, request: ModbusRequest | None = None):
```

Comment:
This looks wrong, why can async_execute have a typed return and execute not ?

Reply:
async_execute needed a explicitly typed return because it also returns bytes in some cases (a, in my opinion, smelly piece of code, but this isnt the PR for that)

the return of execute is not typed explicitly, as type checkers ca infer that from either the mixin or async_execute trivially.

Reply:
that does not make sense !

first of all async_execute should never return bytes but always a ModbusResponse
secondly, you have already declared async_execute as only returning a ModbusResponse 

My preference is clearly if a method is typed, then the whole line ! 

Btw async_execute and execute returns exactly the same...so please point at the place where you think bytes are returned.

Reply:
line 163-165 ihn base.py
```py
if self.broadcast_enable and not request.slave_id:
    resp = b"Broadcast write sent - no response expected"
    break
```[

these are the lines in async_execute which cause there to be a bytes return
i declared async_execute as returning ModbusResponse to allow `type: ignore` to work on the return type. Otherwise mypy would complain a lot about this. I do not know why this code was accepted in the first place

Reply:
> Btw async_execute and execute returns exactly the same

just to be clear: they do not. while async_execute returns a ModbusResponse in most cases, execute returns a Awaitable[ModbusResponse]

in fact, execute returns a Coroutine with the return type ModbusResponse, which is compatible with the Awaitable protocol

Reply:
I am aware that the async returns an awaitable !!! what I meant is that they return the same data, which should always be 
modbusResponse. It still do not explain why execute() is not return typed.

In case of a broadcast there are no response, but I can see we forgot to update base.py to return None, or I suppose it would be better to return an empty ModbusResponse. Please consider that done, I will make a PR doing that later.